cmd/util/ledger/migrations/storage_v4.go

```diff
@@ -66,36 +69,87 @@ func storageFormatV4MigrationWorker(jobs <-chan ledger.Payload, results chan<- s
 		if err != nil {
 			result.error = err
 		} else {
+			if err := checkStorageFormatV4(migratedPayload); err != nil {
+				panic(fmt.Errorf("%w: key = %s", err, payload.Key.String()))
+			}
 			result.Payload = migratedPayload
 		}
 		results <- result
 	}
 }
 
+var decMode = func() cbor.DecMode {
+	decMode, err := cbor.DecOptions{
+		IntDec:           cbor.IntDecConvertNone,
+		MaxArrayElements: math.MaxInt32,
+		MaxMapPairs:      math.MaxInt32,
+		MaxNestedLevels:  256,
+	}.DecMode()
+	if err != nil {
+		panic(err)
+	}
+	return decMode
+}()
+
+var publicKeyKeyPrefix = []byte("public_key_")
+var storageUsedKey = []byte("storage_used")
+var existsKey = []byte("exists")
+var frozenKey = []byte("frozen")
+var uuidKey = []byte("uuid")
+var codeKeyPrefix = []byte("code.")
+var accountAddressStateKey = []byte("account_address_state")
+var contractNamesKey = []byte("contract_names")
+
 func rencodePayloadV4(payload ledger.Payload) (ledger.Payload, error) {
 
-	// If the payload value is not a Cadence value (it does not have the Cadence data magic prefix),
-	// return the payload as is
+	keyParts := payload.Key.KeyParts
+
+	rawOwner := keyParts[0].Value
+	rawKey := keyParts[2].Value
+
+	// Ignore known payload keys that are not Cadence values
+
+	if bytes.HasPrefix(rawKey, publicKeyKeyPrefix) ||
+		bytes.Equal(rawKey, storageUsedKey) ||
+		bytes.Equal(rawKey, existsKey) ||
+		bytes.Equal(rawKey, frozenKey) ||
+		bytes.HasPrefix(rawKey, codeKeyPrefix) ||
+		bytes.HasPrefix(rawKey, accountAddressStateKey) ||
+		bytes.Equal(rawKey, uuidKey) ||
+		bytes.Equal(rawKey, contractNamesKey) {
```

Comment:
While it should be the same functionality I added IsFVMStateKey to state [here](https://github.com/onflow/flow-go/pull/657/files) , so we can use it in other places and migrations.

Reply:
or checks on the equality of owner and controller (in cases that is right) might be an extra measure to make sure we don't have any key collision.

Reply:
refactored in b4657f9